lib/unlock-helpers.js

```diff
@@ -162,8 +160,7 @@ helpers.patternUnlock = async function (adb, driver, capabilities) {
   let piece = size.width / 3;
   let actions = [];
   let lastPos;
-  for (let i in keys) {
-    let key = key[i];
+  for (let key of keys.split(/s*/)) {
     let keyPos = helpers.getPatternKeyPosition(key, pos0, piece);
     if (parseInt(key, 10) === 0) {
```

Comment:
**key** will never be equal to zero according to the algorithm, so this condition is never going to be executed.